velox/functions/prestosql/window/tests/RankTest.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/prestosql/window/tests/WindowTestBase.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class RankTest : public WindowTestBase {};
+
+TEST_F(RankTest, basic) {
+  vector_size_t size = 1000;
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(
+          size, [](auto row) -> int32_t { return row % 10; }),
+      makeFlatVector<int32_t>(
+          size, [](auto row) -> int32_t { return row % 7; }),
+  });
+
+  twoColumnTests({vectors}, "rank");
+}
+
+TEST_F(RankTest, singlePartition) {
+  // Test all input rows in a single partition.
+  vector_size_t size = 1'000;
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(size, [](auto /* row */) { return 1; }),
+      makeFlatVector<int32_t>(size, [](auto row) { return row % 50; }),
+  });
+
+  // Invoking with 2 vectors so that the underlying WindowFunction::apply() is
```

Comment:
Would it make sense to add a helper method to test this use case for a generic window function?

Reply:
I added this case within the generic testTwoColumnInput now.